scripts/checkout_submodules.py

```diff
@@ -69,7 +69,7 @@ def module_matches_platforms(module: Module, platforms: set) -> bool:
     return bool(platforms & module.platforms)
 
 
-def checkout_modules(modules: list, shallow: bool) -> None:
+def checkout_modules(modules: list, shallow: bool, recursive: bool) -> None:
```

Comment:
This is going to checkout a large amount of unnecessary and duplicate code, correct?

Can't you set fetchRecurseSubmodules on the particular gitmodules that need it?

Reply:
in only recurse for modules from the given list (in my case linked with --platform efr32) for others platform the recursive is false.

I can't look into fetchRecurseSubmodules. didn't know about that option

Reply:
It looks like fetchRecurseSubmodules doesn't do what I thought.

Does this break building in the devcontainer with a default checkout ? (i.e., without a manual step to init with --platform efr32) ?

Reply:
umm, it can. You need a recursive checkout of ot-efr32 repo so for thread efr32 examples with this PR.

this is also temporary and should not be needed anymore by the end of the month

Reply:
tested fetchRecurseSubmodules and it indeed doesn't do the recursive checkout of the repo. I will keep the way I did it for now. It is a temporary need anyway and I will change in around 2 weeks.